core/domain/suggestion_services.py

```diff
@@ -593,12 +616,223 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
     return [
         get_suggestion_from_model(suggestion_model) for suggestion_model in (
             suggestion_models.GeneralSuggestionModel
-            .get_translation_suggestions_waiting_longest_for_review_per_lang(
+            .get_translation_suggestions_waiting_longest_for_review(
                 language_code)
         )
     ]
 
 
+def _get_plain_text_from_html_content_string(html_content_string):
+    """Retrieves the plain text from the given html content string. RTE element
+    occurrences in the html are replaced by their corresponding name,
+    capitalized in square brackets.
+    eg: <p>Sample1 <oppia-noninteractive-math></oppia-noninteractive-math>
+        Sample2 </p> will give as output: Sample1 [MATH] Sample2.
+    Note: similar logic exists in the frontend in format-rte-preview.filter.ts.
+
+    Args:
+        html_content_string: str. The content html string to convert to plain
+            text.
+
+    Returns:
+        str. The plain text string from the given html content string.
+    """
+
+    def _replace_rte_tag(rte_tag):
+        """Replaces all of the opening <oppia-noninteractive-**> tags with
+        their corresponding name in square brackets.
+
+        Args:
+            rte_tag: MatchObject. A matched object that contins the
+                oppia-noninteractive rte tag.
+
+        Returns:
+            str. The string to replace the rte tag with.
+        """
+        # Convert the MatchObject to a string.
+        rte_tag_string = rte_tag.group(0)
+        # First grab the **> in the <oppia-noninteractive-**> tag.
```

Comment:
Sorry, no, all this is really still too hacky. Find the entirety of the tag and swap it out, rather than doing it piecemeal like this.

Reply:
I think I made it better Sean :D

Reply:
Looks resolved.